### DISCUSSION TITLE
Lots of serial ports;  how to set up a data structure?

### DISCUSSION THREAD

**Discussion by White-Rabbit-Scientific** [serialport/serialport-rs#212] [Q&A] [answered]
How do I store dyn SerialPort in a struct such that I can have one struct instance per port connection? Or is there a better way?

**Comment by sirhcel** [comment, marked as answer]
Do you mean this like in how to store a serial port instance in a struct? This could be done by storing the serial port trait object returned by `serialport::new()` in a field of this type - `Box<dyn SerialPort>`. For example:

```rust
use serialport::SerialPort;

struct Connection {
    port: Box<dyn SerialPort>,
}

pub fn main() -> anyhow::Result<()> {
    let port = serialport::new("/dev/ttyX", 115200).open()?;
    let mut connection = Connection { port };

    connection.port.write_all(b"Hello world!")?;
    connection.port.flush()?;

    Ok(())
}
```
